python/ray/tests/test_advanced_9.py

```diff
@@ -262,32 +262,32 @@ def test_gcs_connection_no_leak(ray_start_cluster):
     gcs_server_process = head_node.all_processes["gcs_server"][0].process
     gcs_server_pid = gcs_server_process.pid
 
-    ray.init(cluster.address)
-
     def get_gcs_num_of_connections():
         p = psutil.Process(gcs_server_pid)
         print(">>", len(p.connections()))
         return len(p.connections())
 
-    # Wait for everything to be ready.
-    import time
-
-    time.sleep(10)
-
-    fds_without_workers = get_gcs_num_of_connections()
-
     @ray.remote
     class A:
         def ready(self):
             print("HELLO")
             return "WORLD"
 
-    num_of_actors = 10
-    actors = [A.remote() for _ in range(num_of_actors)]
-    print(ray.get([t.ready.remote() for t in actors]))
-
-    # Kill the actors
-    del actors
+    with ray.init(cluster.address):
+        # Wait for everything to be ready.
+        time.sleep(10)
+        # Note: `fds_without_workers` need to be recorded *after* a ray start, because
+        # a prestarted worker is started on the first driver init. This worker keeps 1
+        # connection to the GCS, and it stays alive even after the driver exits. If
```

Comment:
do you know why the connection stays alive even after the driver exits?

Reply:
The prestarted worker is not specifically for job 01000000; it's for job = nil (ffffffff): [code](https://github.com/ray-project/ray/blob/27b6c12e85833ebcd6a640805802868654b29284/src/ray/raylet/worker_pool.cc#L1544). Frankly I don't know what it does - the worker can't be used by any jobs right?

Reply:
Yea, initially the prestarted worker is not tied to a job but I thought later on when we start the actor, it will late bind to job 0100000

Reply:
Oh, those prestarted workers cannot be used for actors

```
void WorkerPool::PrestartDefaultCpuWorkers(ray::Language language, int64_t num_needed) {
  // default workers uses 1 cpu and doesn't support actor.
  static const WorkerCacheKey kDefaultCpuWorkerCacheKey{/*serialized_runtime_env*/ "",
                                                        {{"CPU", 1}},
                                                        /*is_actor*/ false,
                                                        /*is_gpu*/ false};
```